### PR TITLE
feat: update ESLint & TypeScript plugin

### LINKED ISSUE #65: Enjoy the benefits of ESLint's new simplified config
Read the documentation changes in this PR to understand: https://github.com/eslint/eslint/pull/11388/files#diff-33bbe98f532eb757e2789f1a3549c9ca

### LINKED ISSUE #80: Dependencies different from standard
Thanks for this package. Hopefully, they will merge these rules.

Why are the packages required to run this listed as peer dependencies instead of normal dependencies? `standard` just has them listed like this: [standard/package.json#L14-L22](https://github.com/standard/standard/blob/155d52710f587439d58b146f790df322336e195a/package.json#L14-L22)

Also is there a reason that the version requirements are different to the ones standard are using or is just that you're using greenkeeper? It has caused some problems running regression tests against node v6. Namely using version 9 of `eslint-plugin-node` instead of 7 breaks compatibility with node 6.

### LINKED ISSUE #94: peerDependencies should be superset of those of eslint-config-standard

### PR DESCRIPTION
**Pull request by mightyiam**
BREAKING CHANGE: `eslint-config-standard` has been updated to `13.0.1`.
BREAKING CHANGE: Updatd all peer dependencies. Fixes #80. Fixes #94.

Added option for rule `@typescript-eslint/explicit-function-return-type`: `allowHigherOrderFunctions: true`.

BREAKING CHANGE: `typescript-eslint/parser` is no longer a `peerDependency`, but a `dependency`. Users should remove it from their configuration, as well. Closes #65.

Updated `@typescript-eslint/eslint-plugin` to `1.10.2`.

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -1,6 +1,5 @@
 {
   "extends": "./lib/index.js",
-  "parser": "@typescript-eslint/parser",
```
Comment (mightyiam) [inline .eslintrc.json]: https://github.com/eslint/eslint/pull/11388/files#diff-33bbe98f532eb757e2789f1a3549c9caR41

So, our `./lib/index.js` specifies the `parser`.

File: package.json [hunk 2]
```diff
@@ -52,25 +52,25 @@
     "TypeScript"
   ],
   "dependencies": {
-    "eslint-config-standard": "^12.0.0"
+    "@typescript-eslint/parser": "^1.13.0",
```
Comment (mightyiam) [inline package.json]: https://github.com/eslint/eslint/pull/11388/files#diff-33bbe98f532eb757e2789f1a3549c9caR41

The parser can and should be a `dependency`. Makes it easier for the user, right? #65.

A parser update seems necessary for ESLint 6. So I updated it to the latest version.

File: package.json [hunk 3]
```diff
@@ -52,25 +52,25 @@
...
-    "eslint-plugin-promise": "^4.0.1",
-    "eslint-plugin-standard": "^4.0.0",
-    "@typescript-eslint/eslint-plugin": "^1.4.2",
-    "@typescript-eslint/parser": "^1.4.2"
+    "eslint": ">=6.0.1",
+    "eslint-config-standard": ">=13.0.1",
+    "eslint-plugin-import": ">=2.18.0",
+    "eslint-plugin-node": ">=9.1.0",
+    "eslint-plugin-promise": ">=4.2.1",
+    "eslint-plugin-standard": ">=4.0.0",
```
Comment (mightyiam) [inline package.json]: #94, #80.

https://github.com/standard/eslint-config-standard/blob/f7c0fe1078754ab0c38c69df675fbacfbbd96178/package.json#L50-L56

File: package.json [hunk 4]
```diff
@@ -52,25 +52,25 @@
...
   },
   "devDependencies": {
     "@commitlint/cli": "^8.1.0",
     "@commitlint/config-conventional": "^8.1.0",
     "@commitlint/travis-cli": "^8.1.0",
     "@types/node": "^12.6.9",
-    "@typescript-eslint/eslint-plugin": "^1.9.0",
-    "@typescript-eslint/parser": "^1.11.0",
-    "eslint": "^5.16.0",
+    "@typescript-eslint/eslint-plugin": "^1.10.2",
```
Comment (mightyiam) [inline package.json]: I did not intend to update the TypeScript plugin. Yet, the current version seems to crash with some error when using the latest parser.

File: package.json [hunk 5]
```diff
@@ -52,25 +52,25 @@
...
+    "eslint": "^6.1.0",
```
Comment (mightyiam) [inline package.json]: To match @standard's version.

File: package.json [hunk 6]
```diff
@@ -52,25 +52,25 @@
...
+    "eslint-plugin-standard": ">=4.0.0",
+    "@typescript-eslint/eslint-plugin": ">=1.10.1"
   },
   "devDependencies": {
     "@commitlint/cli": "^8.1.0",
     "@commitlint/config-conventional": "^8.1.0",
     "@commitlint/travis-cli": "^8.1.0",
     "@types/node": "^12.6.9",
-    "@typescript-eslint/eslint-plugin": "^1.9.0",
-    "@typescript-eslint/parser": "^1.11.0",
```
Comment (mightyiam) [inline package.json]: Moved to `dependencies` and updated.

File: readme.md [hunk 7]
```diff
@@ -10,23 +10,21 @@ An [ESLint shareable config](https://eslint.org/docs/developer-guide/shareable-c
 ## Usage
 
 ```
-npm install --save-dev eslint-plugin-standard eslint-plugin-promise eslint-plugin-import eslint-plugin-node @typescript-eslint/parser @typescript-eslint/eslint-plugin eslint-config-standard-with-typescript 
+npm install --save-dev eslint-plugin-standard eslint-plugin-promise eslint-plugin-import eslint-plugin-node @typescript-eslint/eslint-plugin
```
Comment (mightyiam) [inline readme.md]: Both the config we extend from and the parser are `dependencies` of this package, so users should not install them.
#65.

File: readme.md [hunk 8]
```diff
@@ -10,23 +10,21 @@ An [ESLint shareable config](https://eslint.org/docs/developer-guide/shareable-c
...
 ```
 
-Yes, I know it is a large number of packages. This is due to [a known design flaw in ESLint](https://github.com/eslint/eslint/issues/10125).
+Yes, I know it is a large number of packages. This is due to [a known limitation in ESLint](https://github.com/eslint/eslint/issues/3458).
```
Comment (mightyiam) [inline readme.md]: Kind of related.

File: readme.md (hunk 7 above)
Comment (feross) [reply readme.md]: This is a nice improvement. Lowers the number of packages the user has to install.

**Comment by feross** [comment]
I removed `package-lock.json` from the repo because it's counterproductive for module authors.  https://github.com/standard/eslint-config-standard-with-typescript/commit/51c7645f1eafaf08fb85151ec6f0ecf1a21eab34

So I updated this PR to fix the merge conflict. :)

**Comment by feross** [comment]
There seems to be a test failure.

**Comment by mightyiam** [comment]
In favor of #105.

**Pull request closed without merging**